lua/nudge-two-hats/buffer.lua

```
@@ -293,17 +309,20 @@ function M.get_prompt_for_buffer(buf, state)
             print("[Nudge Two Hats Debug] Selected hat: " .. selected_hat)
           end
         end
-        return string.format("I am a %s wearing the %s hat. %s. With %s emotions and a %s tone, I will advise: %s", 
+        local base = string.format("I am a %s wearing the %s hat. %s. With %s emotions and a %s tone, I will advise: %s",
                              role, selected_hat, direction, emotion, tone, prompt_text)
+        return (base .. " " .. cb_result):gsub("%s+$", "")
       else
         selected_hat = nil
-        return string.format("I am a %s. %s. With %s emotions and a %s tone, I will advise: %s", 
+        local base = string.format("I am a %s. %s. With %s emotions and a %s tone, I will advise: %s",
                              role, direction, emotion, tone, prompt_text)
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'role'

(W113)